src/keymap.rs

```diff
@@ -0,0 +1,416 @@
+use config::Config;
+use config::EditMode;
+use consts::KeyPress;
+use tty::RawReader;
+use super::Result;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Cmd {
+    Abort, // Miscellaneous Command
+    AcceptLine,
+    BackwardChar(i32),
+    BackwardDeleteChar(i32),
+    BackwardKillWord(i32, Word), // Backward until start of word
+    BackwardWord(i32, Word), // Backward until start of word
+    BeginningOfHistory,
+    BeginningOfLine,
+    CapitalizeWord,
+    ClearScreen,
+    Complete,
+    DeleteChar(i32),
+    DowncaseWord,
+    EndOfFile,
+    EndOfHistory,
+    EndOfLine,
+    ForwardChar(i32),
+    ForwardSearchHistory,
+    ForwardWord(i32, At, Word), // Forward until start/end of word
+    Interrupt,
+    KillLine,
+    KillWholeLine,
+    KillWord(i32, At, Word), // Forward until start/end of word
+    NextHistory,
+    Noop,
+    PreviousHistory,
+    QuotedInsert,
+    Replace(i32, char), // TODO DeleteChar + SelfInsert
+    ReverseSearchHistory,
+    SelfInsert(char),
+    Suspend,
+    TransposeChars,
+    TransposeWords,
+    Unknown,
+    UnixLikeDiscard,
+    // UnixWordRubout, // = BackwardKillWord(Word::Big)
+    UpcaseWord,
+    ViCharSearch(i32, CharSearch),
+    ViDeleteTo(i32, CharSearch),
+    Yank(i32, Anchor),
+    YankPop,
+}
+
+#[derive(Debug, Clone, PartialEq, Copy)]
+pub enum Word {
+    // non-blanks characters
+    Big,
+    // alphanumeric characters
+    Emacs,
+    // alphanumeric (and '_') characters
+    Vi,
+}
+
+#[derive(Debug, Clone, PartialEq, Copy)]
+pub enum At {
+    Start,
+    End,
+}
+
+#[derive(Debug, Clone, PartialEq, Copy)]
+pub enum Anchor {
```

Comment:
I chose this word because 'where' is a Rust keyword...
Any suggestion is welcome.